cognite_synthetic_tags/tag.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+from typing import Any, Optional
+
+from ._operations import INFIX_OPERATORS
+from .types import OperatorT, TagFormulaT
+
+
+class Tag:
+    def __init__(self, name: str, store: Optional[str] = None):
+        self.name = f"{name}__{store}" if store else name
+        self.store = store
+        self._value = None
+        self.formula: Optional[TagFormulaT] = None
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __repr__(self):
+        return self.__str__()
+
+    def calc(self, operator_: OperatorT, *args: Any) -> Tag:
+        return Tag.call(operator_, self, *args)
+
+    @classmethod
+    def call(cls, operator_: OperatorT, *args: Any) -> Tag:
+        if callable(operator_):
+            oper_str = operator_.__name__
+        else:
+            oper_str = str(operator_)
+            if oper_str.startswith("r"):
+                args = args[::-1]
+        if oper_str in INFIX_OPERATORS:
+            oper_str = INFIX_OPERATORS[oper_str]
+            new_tag = Tag(f"{f' {oper_str} '.join(map(str, args))}")
+        else:
+            new_tag = Tag(f"{oper_str}({', '.join(map(str, args))})")
+        new_tag.formula = (operator_, args)
+        return new_tag
+
+    # forward binary operations, e.g: Tag + 3
+
+    def __add__(self, other: Any) -> Tag:
```

Comment:
Can the type of `other` be anything else than `float` or `Tag` ?

Reply:
Yes, for example a string. But also I don't want to limit use of more exotic stuff like [`LazyString`](https://github.com/oblalex/lazy-string).